src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -21,282 +26,459 @@ data class KmerMapData(val rangeToBitSetMap: Map<ReferenceRange, BitSet>, val km
+                } else {
+                    null
+                }
 
+                //Setting up the channels for the coroutines
+                val readStringChannel = Channel<Pair<String, String>>(100)
+                val sortedHapidListChannel = Channel<List<String>>(100)
+                val numberOfMappingThreads = max(1, numThreads - 2)
 
-/**
- * Function to process a read or a readPair into their corresponding hapIdSet.  This will work with both single and paired end reads.
- * To Work with single ended reads, the second String in the Pair should be an empty String.
- */
-suspend fun processReads(reads: ReceiveChannel<Pair<String, String>>,
-                               sortedLists: SendChannel<List<Int>>,
-                               minProportionOfMaxCount: Double = 1.0,
-                               minSameReferenceRange: Double = 0.9,
-                               kmerHashOffsetMap: Long2LongOpenHashMap,
-                               refrangeToBitSet: Map<Int, BitSet>,
-                               rangeToHapidIndexMap: Map<Int, Map<Int, Int>>) {
-    for (pairOfReads in reads) {
-        val result1 = readToHapidSet(pairOfReads.first, minProportionOfMaxCount, minSameReferenceRange, kmerHashOffsetMap, refrangeToBitSet, rangeToHapidIndexMap)
-        val result2 = if(pairOfReads.second.isNotEmpty()) {
-            readToHapidSet(pairOfReads.second, minProportionOfMaxCount, minSameReferenceRange, kmerHashOffsetMap, refrangeToBitSet, rangeToHapidIndexMap)
-        }
-        else {
-            setOf<Int>()
+                //Run the blocking coroutine to read the fastq files and process the reads
+                runBlocking {
+                    launch(Dispatchers.IO) {
+                        while (reader1.hasNext() && reader2?.hasNext() ?: true) {
+                            val seq1 = reader1.next().readString
+                            val seq2 = reader2?.next()?.readString ?: ""
+
+                            readStringChannel.send(Pair(seq1, seq2))
+                            readCount++
+
+                        }
+                        readStringChannel.close()
+                    }
+
+                    val jobList: MutableList<Job> = mutableListOf()
+                    repeat(numberOfMappingThreads) {
+                        jobList.add(launch(Dispatchers.Default) {
+                            processReads(
+                                readStringChannel,
+                                sortedHapidListChannel,
+                                minProportionOfMaxCount,
+                                minSameReferenceRange,
+                                kmerIndexMap.kmerHashToLongMap,
+                                rangeIdToBitsetMap,
+                                graph.refRangeIdToHapIdMap()
+                            )
+                        })
+                    }
+
+                    val listJob = launch {
```

Comment:
jobList and listJob are non-informative names. Maybe something a bit more descriptive? readJobList and addJobList?